applications/StructuralMechanicsApplication/custom_processes/compute_moment_of_inertia_process.cpp

```diff
@@ -0,0 +1,65 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:             BSD License
+//                                       license: StructuralMechanicsApplication/license.txt
+//
+//  Main authors:    Philipp Bucher
+//                   Salman Yousaf
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "custom_processes/compute_moment_of_inertia_process.h"
+#include "custom_processes/total_structural_mass_process.h"
+#include "structural_mechanics_application_variables.h"
+namespace Kratos
+{
+
+void ComputeMomentOfInertiaProcess::Execute()
+{
+    KRATOS_TRY
+
+    const std::size_t domain_size = mrThisModelPart.GetProcessInfo()[DOMAIN_SIZE];
+    double moment_of_inertia = 0.0;
+
+    // computing axis of rotation
+    Vector3 axis_of_rotation = mrPoint2 - mrPoint1;
+    const double axis_length =  std::sqrt(inner_prod(axis_of_rotation, axis_of_rotation));
+    KRATOS_ERROR_IF(axis_length < 1e-12) << "Please specify two different points" << std::endl;
+
+    // Now we iterate over the elements
+    auto& elements_array = mrThisModelPart.GetCommunicator().LocalMesh().Elements();
+    // Making this loop omp-parallel requires locking all the geometries & nodes, which
+    // is most probably not worth the effort
+    for (auto& elem_i : elements_array) {
+        const double elem_mass = TotalStructuralMassProcess::CalculateElementMass(elem_i, domain_size);
```

Comment:
Should be this an external utility?, common for this process and TotalStructuralMassProcess

Reply:
maybe in the future we can move it ...

Should be a free function ideally